app/models/programme_activity_grouping.rb

```diff
@@ -15,14 +16,12 @@ def achievements(user)
   end
 
   def user_complete?(user)
-    completed_activity_count = 0
-    user_achievements = achievements(user).to_a
-    programme_activities.each do |programme_activity|
-      completed_activity = user_achievements.find { _1.activity_id == programme_activity.activity_id }
-      completed_activity_count += 1 if completed_activity
-      return true if completed_activity_count >= required_for_completion
-    end
-    nil
+    user_achievement_activity_ids = achievements(user).pluck(:activity_id)
+
+    completed_activity_count = programme_activities
+      .count { _1.activity_id.in? user_achievement_activity_ids }
+
+    completed_activity_count >= required_for_completion
```

Comment:
Is `#user_complete?` a refactor with no functional changes? If not what changed here?

Reply:
It's a refactor with next to no functional changes. It returns false instead of nil when It's "not true". This doesn't affect any of the locations in which it is used though